File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.java
class Solution {
    public String reverseVowels(String s) 
    {
        char[] arr=s.toCharArray();
        int l=0;
        int r=arr.length-1;
        while(l<r)
        {
            while(l<r && !isVowel(arr[l]))
            {
                l++;
            }
            while(l<r && !isVowel(arr[r]))
            {
                r--;
            }
            char temp=arr[l];
            arr[l]=arr[r];
            arr[r]=temp;
            l++;
            r--;
        }
        return new String(arr);
    }
    private boolean isVowel(char ch) 
    {
        return "aeiouAEIOU".indexOf(ch) != -1;
}
}

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        c=list(s)
        l=0
        r=len(c)-1;
        v="aeiouAEIOU"
        while(l<r):
            while(l<r and c[l] not in v):
                l+=1
            while(l<r and c[r] not in v):
                r-=1
            c[l],c[r]=c[r],c[l]
            l+=1
            r-=1

        return "".join(c)

        